api/valor_api/backend/metrics/text_generation.py

```diff
@@ -0,0 +1,529 @@
+                datum_subquery.c.datum_text.label("datum_text"),
+                prediction_subquery.c.prediction_text.label("prediction_text"),
+                prediction_subquery.c.prediction_context.label(
+                    "prediction_context"
+                ),
+            )
+            .select_from(datum_subquery)
+            .join(
+                prediction_subquery,
+                datum_subquery.c.datum_id == prediction_subquery.c.datum_id,
+            )
+        )
+
+        results = db.execute(total_query).all()
+        is_AnswerRelevance_enabled = "AnswerRelevance" in metrics_to_return
+        is_Coherence_enabled = "Coherence" in metrics_to_return
+
+        for datum_uid, dataset_name, datum_text, prediction_text, _ in results:
+            if is_AnswerRelevance_enabled:
+                score = client.answer_relevance(
+                    query=datum_text, text=prediction_text
+                )
+                output += [
+                    schemas.AnswerRelevanceMetric(
+                        value=score,
+                        parameters={
+                            "dataset": dataset_name,
+                            "datum_uid": datum_uid,
+                            "prediction": prediction_text,
+                        },
+                    )
+                ]
+            if is_Coherence_enabled:
+                generated_text = prediction_text
+                score = client.coherence(text=generated_text)
+                output += [
+                    schemas.CoherenceMetric(
+                        value=score,
+                        parameters={
+                            "dataset": dataset_name,
+                            "datum_uid": datum_uid,
+                            "prediction": prediction_text,
+                        },
+                    )
+                ]
+
+    return output
+
+
+@validate_computation
+def compute_text_generation_metrics(
+    *,
+    db: Session,
+    evaluation_id: int,
+) -> int:
+    """
+    Compute text generation metrics. This function is intended to be run using FastAPI's `BackgroundTasks`.
+
+    Parameters
+    ----------
+    db : Session
+        The database Session to query against.
+    evaluation_id : int
+        The job ID to create metrics for.
+
+    Returns
+    ----------
+    int
+        The evaluation job id.
+    """
+
+    # fetch evaluation
+    evaluation = core.fetch_evaluation_from_id(db, evaluation_id)
+
+    # unpack filters and params
+    datum_filter = schemas.Filter(**evaluation.filters)
+    prediction_filter = datum_filter.model_copy()
+    groundtruth_filter = datum_filter.model_copy()
+    parameters = schemas.EvaluationParameters(**evaluation.parameters)
+
+    log_evaluation_item_counts(
+        db=db,
+        evaluation=evaluation,
+        prediction_filter=prediction_filter,
+        groundtruth_filter=groundtruth_filter,
+    )
+
+    assert parameters.metrics_to_return
```

Comment:
`assert` should not be used in production code as it can be ignored depending on python interpreter flags. See [link](https://stackoverflow.com/questions/944592/best-practice-for-using-assert).

This should be changed to an explicit "if x then raise"